v1/submissions.go

```diff
@@ -250,3 +475,57 @@ func (e *Endpoint) GetSubmissions() ([]Submission, error) {
 
 	return findings, nil
 }
+
+/*
+	GetSubmission gets report/submission by code/id
+*/
+func (e *Endpoint) GetSubmission(code string) (*SubmissionDetails, error) {
+	var submi SubmissionDetails
+	var respBytes []byte
+	var err error
+	var req *http.Request
+	url := apiEndpointV1 + "/SCHIBSTED-LKJ2GIWL"
+
+	if err := authenticate(e); err != nil {
+		return nil, errors.Wrap(err, "could not authenticate to intigriti API")
+	}
+
+	req, err = http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create http request to intigriti")
+	}
+	req.Header.Set("Content-Type", mimeFormUrlEncoded)
+	req.Header.Set("X-Client", clientTag)
+	req.Header.Set("Authorization", getBearerTokenHeader(e.authToken))
+
+	client := http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "fetching to intigriti failed")
+	}
+
+	defer func() { _ = resp.Body.Close() }()
+
+	// the token was invalidated before it expired
+	if resp.StatusCode == http.StatusUnauthorized {
+		// fetch a new token
+		if err := authenticate(e); err != nil {
+			return nil, errors.Wrap(err, "could not reauthenticate to intigriti API")
+		}
+		e.Logger.Debug("reauthenticated because of an invalidated token")
+	}
+
+	if resp.StatusCode > 399 {
+		return nil, errors.Errorf("fetch from intigriti returned status code: %d", resp.StatusCode)
+	}
+
+	respBytes, err = ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not read response")
+	}
+	err = json.Unmarshal(respBytes, &submi)
+	if err != nil && submi.Code == "" {
```

Comment:
what is this `submi.Code` special case? if the err is not nil, then submi will be empty?

Reply:
I got an error for a specific report on the substructure of SubmissionDetails for the field Events in which the subfield To was different or empty than the one in the structure. The rest of the structure was unmarshalled correctly. 

I didn't dig up more to find a better structure that fits this case. From experience, it seems we'll have to use an interface for the events and declare a separate structure in which we can map the new To interface. That will change many things and the workflow